### PR TITLE
feat: send confirmation and verification code

### PR DESCRIPTION
**Pull request by Davidmattei**
|Q              |A  |
|---------------|---|
|Bug fix?       |no|
|New feature?   |yes|	
|BC breaks?     |no|	
|Deprecations?  |no|	
|Fixed tickets  |no|	

- New validation: **EMS\FormBundle\Components\Validation\VerificationCode**
- New field: **EMS\FormBundle\Components\Field\SendConfirmation**
- New configuration: endpoints: '%env(json:EMSF_ENDPOINTS)%'

```json
[
  {
    "field_name": "send_confirmation",
    "http_request": {
      "url": "https://api.example.test/v1/send/sms",
      "headers": {
        "Content-Type": "application/json"
      },
      "body": "{\"To\": \"%value%\", \"Message\": \"%verification_code%\"}"
    }
  }
]
```

### CODE REVIEW

File: Resources/views/form_theme.html.twig [hunk 1]
```diff
@@ -31,4 +31,18 @@
...
+{%- endblock -%}
+
+{%- block ems_send_confirmation_widget -%}
+    <div class="input-group">
+        <div class="input-group-append">
+            <button class="btn btn-success btn-send-confirmation" type="button"
+                    data-token="{{ csrf_token(token_id) }}"
+                    data-name="{{ name }}"
+                    data-value-id="{{ attribute(form.parent.children, value_field).vars.id }}">
+                {{ 'ems_send_confirmation'|trans({}, ems_translation_domain) }}
```
Comment (theus77) [inline Resources/views/form_theme.html.twig:43]: So we are not able to easily translate this button?

**Review by theus77 — DISMISSED** [review]

File: Service/Confirmation/ConfirmationService.php [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+            $verificationCode = $this->verificationService->create($confirmationRequest->getValue());
+
+            if (null === $verificationCode) {
+                throw new \Exception('Failed getting verification code');
+            }
+
+            $body = $request->createBody($confirmationRequest->getValue(), $verificationCode);
+
+            $response = $this->httpClient->request($request->getMethod(), $request->getUrl(), [
```
Comment (Davidmattei) [inline Service/Confirmation/ConfirmationService.php:63]: valid response content, if sms is really send!

File: DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -20,6 +20,9 @@ public function getConfigTreeBuilder()
         $rootNode
             ->children()
                 ->variableNode('hashcash_difficulty')->defaultValue(16384)->end()
+                ->variableNode('endpoints')
```
Comment (Davidmattei) [inline DependencyInjection/Configuration.php:23]: Update documentation

File: Components/Constraint/IsVerificationCode.php [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+namespace EMS\FormBundle\Components\Constraint;
+
+use Symfony\Component\Validator\Constraint;
+
+final class IsVerificationCode extends Constraint
+{
+    /** @var string */
+    public $field;
+    /** @var string */
+    public $message = 'The verification code "{{code}}" is not valid.';
```
Comment (Davidmattei) [inline Components/Constraint/IsVerificationCode.php:14]: Add default translation for NL and FR and test EN DE

File: DependencyInjection/Configuration.php (hunk 3 above)
Comment (Davidmattei) [reply DependencyInjection/Configuration.php:23]: 95df8d8

File: Service/Confirmation/ConfirmationService.php (hunk 2 above)
Comment (Davidmattei) [reply Service/Confirmation/ConfirmationService.php:63]: 787aeaa

**Review by theus77 — APPROVED** [review]

**Pull request merged**